ztp/ztp-policy-generator/README.md

```diff
@@ -0,0 +1,318 @@
+
+1 - Policy Categorization:
+
+ Any generated policy needs to be placed under one of the following categories; 
+
+   - Common: a policy exist under the common category will be applied to all clusters.
+
+   - Groups: a policy exist under the group category will be applied to group of clusters.
+
+   - Sites: a policy exist under the sites category will be applied to a specific cluster.
+
+2 - Source CR policy:
+
+ The CR that will be used to generate the ACM policy needs to be defined accordingly to the category that will be set to it. 
+For example; a common-namespace-policy contain a Namespace definition that will exist in all managed clusters. This namespace will be placed under the common category and there will be no changes for its spec or data across all clusters. The source CR for this namespace will be as below
+
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+ name: openshift-sriov-network-operator
+ labels:
+   openshift.io/run-level: "1"
+```
+
+The generated policy that will apply this namespace will include the namespace as it is defined above without any change. It will be as below 
+
+```
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+   name: common-sriov-sub-ns-policy
+   namespace: common-sub
+   annotations:
+       policy.open-cluster-management.io/categories: CM Configuration Management
+       policy.open-cluster-management.io/controls: CM-2 Baseline Configuration
+       policy.open-cluster-management.io/standards: NIST SP 800-53
+spec:
+   remediationAction: enforce
+   disabled: false
+   policy-templates:
+       - objectDefinition:
+           apiVersion: policy.open-cluster-management.io/v1
+           kind: ConfigurationPolicy
+           metadata:
+               name: common-sriov-sub-ns-policy-config
+           spec:
+               remediationAction: enforce
+               severity: low
+               namespaceselector:
+                   exclude:
+                       - kube-*
+                   include:
+                       - '*'
+               object-templates:
+                   - complianceType: musthave
+                     objectDefinition:
+                       apiVersion: v1
+                       kind: Namespace
+                       metadata:
+                           labels:
+                               openshift.io/run-level: "1"
+                           name: openshift-sriov-network-operator
+```
+
+Another example siteN-sriov-nnp-policy; a SriovNetworkNodePolicy definition that will be exist in different clusters with different spec for each cluster. The source CR for the SriovNetworkNodePolicy will be as below
+
+```
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: sriov-nnp
+  namespace: openshift-sriov-network-operator
+spec:
+  deviceType: $deviceType
```

Comment:
Maybe move the note below on use of "$xxxx" variables up here. The use of "$xxxxx" variables in the example source CRs makes it look like there is a variable substitution but as I understand it PolicyGen is doing an overlay in a way similar to Kustomize.

Reply:
okay